addons/web/static/src/core/tree_editor/condition_tree.js

```diff
@@ -773,6 +847,78 @@ export function removeBetweenOperators(tree) {
     return newTree;
 }
 
+export function removeWithinOperators(tree) {
+    if (tree.type === "complex_condition") {
+        return tree;
+    }
+    if (tree.type === "condition") {
+        if (tree.operator !== "within") {
+            return tree;
+        }
+
+        const { negate, path, value } = tree;
+        const dateType = value[2];
+
+        if (!dateType) {
+            return tree;
+        }
+
+        if (dateType === "date") {
+            return condition(
+                path,
+                "between",
+                [
+                    value[0] > 0
+                        ? expression(`context_today().strftime("%Y-%m-%d")`)
```

Comment:
you can avoid duplication here. For example,
you could create an array
and reverse it when needed